WordPress/Classes/ViewRelated/Site Creation/Verticals/VerticalErrorRetryTableViewCell.swift

```diff
@@ -0,0 +1,142 @@
+
+import UIKit
+
+import Gridicons
+import WordPressShared
+
+// MARK: - VerticalErrorRetryTableViewCellAccessoryView
+
+/// The accessory view comprises a retry Gridicon
+private class VerticalErrorRetryTableViewCellAccessoryView: UIStackView {
+
+    // MARK: Properties
+
+    /// A collection of parameters uses for view layout
+    private struct Parameters {
+        static let minimumHeight    = CGFloat(28)
+        static let retryDimension   = CGFloat(16)
+        static let padding          = CGFloat(4)
+    }
+
+    /// One of the arranged subviews : a "refresh" Gridicon
+    private let retryImageView: UIImageView
+
+    /// One of the arranged subviews : user-facing text
+    private let retryLabel: UILabel
+
+    // MARK: VerticalErrorRetryTableViewCellAccessoryView
+
+    init() {
+        self.retryImageView = {
+            let dismissImage = Gridicon.iconOfType(.refresh).imageWithTintColor(WPStyleGuide.wordPressBlue())
+            let imageView = UIImageView(image: dismissImage)
+
+            imageView.translatesAutoresizingMaskIntoConstraints = false
+            imageView.contentMode = .scaleAspectFit
+
+            return imageView
+        }()
+
+        self.retryLabel = {
+            let label = UILabel()
+
+            label.translatesAutoresizingMaskIntoConstraints = false
+
+            label.font = WPStyleGuide.fontForTextStyle(.body, fontWeight: .bold)
+            label.textColor = WPStyleGuide.wordPressBlue()
+            label.textAlignment = .center
+
+            label.text = NSLocalizedString("Retry", comment: "Title for accessory view in the empty state table view cell in the Verticals step of Enhanced Site Creation")
+            label.sizeToFit()
+
+            return label
+        }()
```

Comment:
I wonder. Why do you choose to return these (retryLabel and retryImageView) inside the init method as appose to having them be computed properties (maybe even lazy once)?

Reply:
I like this approach because it offers the benefit of immutability.

As for `lazy var`, I used to prefer that, but it is not thread-safe. From [_The Swift Programming Language Guide_](https://docs.swift.org/swift-book/LanguageGuide/Properties.html):

_If a property marked with the lazy modifier is accessed by multiple threads simultaneously and the property has not yet been initialized, there is no guarantee that the property will be initialized only once._

Reply:
Ahh. I get it. Hopefully only the main thread would access these UI elements but better safe... 😬